src/syntax/syntax-check.rkt

```diff
@@ -47,10 +52,13 @@
     [#`(#,f-syntax #,args ...)
      (define f (syntax->datum f-syntax))
      (if (hash-has-key? parametric-operators f)
-         (unless (or (symbol? (get-operator-itype f))
-                     (= (length args) (length (get-operator-itype f))))
-           (error! stx "Operator ~a given ~a arguments (expects ~a)"
-                   f (length args) (length (get-operator-itype f))))
+         (let ([arity (get-operator-arity f)])
+           (unless (within (length args) arity)
+             (if (= (car arity) (cdr arity))
+                 (error! stx "Operator ~a given ~a arguments (expects ~a)"
+                         f (length args) (car arity))
+                 (error! stx "Operator ~a given ~a arguments (expects [~a, ~a])"
+                         f (length args) (car arity) (cdr arity)))))
```

Comment:
What's an example operator that can take different numbers of arguments? (Except the variary operators, anyway.)

Reply:
The only case is `neg` vs `-`. This is already handled though, so I'm not sure if this is necessary.

Reply:
I think it's better if we _don't_ have a mechanism for this, and just change `expand-associativity` to create abstract `-` and `neg` operators. And special-case it in the type checker if we need to.